### DISCUSSION TITLE
What does borg info displays, exactly?

### DISCUSSION THREAD

**Discussion by wsw70** [borgbackup/borg#6927] [Q&A] [answered]
The [documentation ](https://borgbackup.readthedocs.io/en/stable/usage/info.html) for `borg info` states that

> Please note that the deduplicated sizes of the individual archives do not add up to the deduplicated size of the repository (“all archives”), because the two are meaning different things:
> This archive / deduplicated size = amount of data stored ONLY for this archive = unique chunks of this archive. All archives / deduplicated size = amount of data stored in the repo = all chunks in the repository.

I do not understand these numbers vs the size of the backup directory:

```
root@srv /b/b/srv# /usr/bin/borg info /backup/borg/srv
Repository ID: c6853ce826e2c5c62c58df2ac0eabca4cf087980944b57995703bd44cceb981e
Location: /backup/borg/srv
Encrypted: No
Cache: /root/.cache/borg/c6853ce826e2c5c62c58df2ac0eabca4cf087980944b57995703bd44cceb981e
Security dir: /root/.config/borg/security/c6853ce826e2c5c62c58df2ac0eabca4cf087980944b57995703bd44cceb981e
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
All archives:                7.01 TB              6.40 TB            105.04 GB

                       Unique chunks         Total chunks
Chunk index:                  197396             10418354

root@srv /b/b/srv# du -sh  /backup/borg/srv
665G    /backup/borg/srv
```

The space taken by the backup is 665 GB. What is its relationship to 

- 7.01 TB → I think this is the size that is actually backed up, in other words, if I recovered all backups separately they would take in total 7.01 TB
- 6.40 TB → the same as above, compressed (I am not sure what is interesting in that number)
- 105.04 GB → I expected this to be the actual size of the backup *on the disk*, because thanks to deduplication, files/chunks in the 7 TB above can be resued between the backups, and the 105 GB is just the "net backup" (files that are either unique, or can be pointed to several times in the different backups, but need to be stored only once)

EDIT:
```
root@srv ~# borg -V
borg 1.2.1
```

**Comment by infectormp** [comment]
See comments here https://github.com/borgbackup/borg/issues/5038

**Reply by wsw70** [reply]
Well, I am not sure this is really the case. From https://github.com/borgbackup/borg/issues/5038#issuecomment-599491540:

> Deduplicated size = size of backup after deduplication. This is size of your repository on a file system.

In my case duplicated size is 105 GB and the size on disk is 665 GB. This is a factor of 6, so something is not right with the numbers.

**Reply by ThomasWaldmann** [reply]
The top post is missing the version information from `borg -V`.

**Reply by wsw70** [reply]
Ah yes, this is 1.2.1, I will update

**Comment by infectormp** [comment, marked as answer]
Sorry i overlooked your question about space
Try to use [borg compact](https://borgbackup.readthedocs.io/en/stable/usage/compact.html#borg-compact) command and then check space again

**Reply by ThomasWaldmann** [reply]
Note that `borg compact` only exists for borg >= 1.2.0 (older versions did always compact and did not have a separate command for it).

Also check your `repo_dir/config` whether it has `append_only` activated.

**Reply by wsw70** [reply]
I have the default `append_only = 0` setting.

After `borg compact` the size on disk went down to 100 GB, aligned with what is reported by borg.

I will run this from time to time, especially when reorganizing my backup (and removing quite a lot of data).

**Reply by ThomasWaldmann** [reply]
the fact that one has to run `borg compact` to reduce disk usage is clearly pointed out in the docs as well as in the borg 1.2 changelog.

you should run it regularly. if you back up daily, you could e.g. run it daily or weekly.

if you do not run it for a longer time, disk usage will increase and also the time needed to free disk space will increase.
